src/WpfMath/TexFormulaParser.cs

```diff
@@ -393,20 +395,58 @@ private TexFormula ReadScript(TexFormula formula, SourceSpan value, ref int posi
 
                     source = value.Segment(start, sqrtEnd - start);
                     return new Radical(source, sqrtFormula.RootAtom, degreeFormula?.RootAtom);
-
-                case "color":
+                    }
+                case "fgcolor":
                     {
+                        //Command to change the foreground color 
                         var colorName = ReadGroup(formula, value, ref position, leftGroupChar, rightGroupChar);
-                        var remainingString = value.Segment(position);
+
+                        string remainingString = value.Substring(position);
                         var remaining = Parse(remainingString, formula.TextStyle);
                         position = value.Length;
-                        if (predefinedColors.TryGetValue(colorName.ToString(), out var color))
+
+                        if (predefinedColors.TryGetValue(colorName, out Color color))
                         {
-                            source = value.Segment(start, position - start);
-                            return new StyledAtom(source, remaining.RootAtom, null, new SolidColorBrush(color));
+                            return new StyledAtom(remaining.RootAtom, null, new SolidColorBrush(color));
+                        }
+                        else
+                        {
+                            try
+                            {
+                                Color color1 = UserDefinedColorParser.ParseUserColor(colorName);
+                                return new StyledAtom(remaining.RootAtom, null, new SolidColorBrush(color1));
+                            }
+                            catch
```

Comment:
I may not be able to change the code now(it would take a lot of time), since my internet service is poor and Visual Studio is massive. I would really appreciate it if you could make the needed changes.

Reply:
Not a problem, I could fix that.

Reply:
Thank you very much

Reply:
Okay, I've fixed the compilation issue.

Reply:
Thank you